File: com/pdfjet/B5.java
/*
 * B5.java
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.pdfjet;

/**
 * Used to specify PDF page with size <strong>B5</strong>, the ISO 216 B5 of 176 by 250 mm.
 * For more information about the page size classes - A3, A4, A5, B5, JISB5, Executive, Letter, Legal and Tabloid - see the Page class.
 */
public class B5 {
    /** The default constructor */
    public B5() {
    }
    /**
     * This is a public static variable that specifies that page size in portrait orientation.
     */
    public static final PageSize PORTRAIT = new PageSize(499.0f, 709.0f);
    /**
     * This is a public static variable that specifies that page size in landscape orientation.
     */
    public static final PageSize LANDSCAPE = new PageSize(709.0f, 499.0f);
}

File: src/b5/b5.go
// b5.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

// Package b5 defines the B5 page size, the ISO 216 B5 of 176 by 250 mm, in points.
package b5

import "github.com/edragoev1/pdfjet/v9/src/pagesize"

// Portrait returns the B5 page size in portrait orientation.
func Portrait() pagesize.PageSize {
	return pagesize.NewPageSize(499.0, 709.0)
}

// Landscape returns the B5 page size in landscape orientation.
func Landscape() pagesize.PageSize {
	return pagesize.NewPageSize(709.0, 499.0)
}
